validation/validate.py

```diff
@@ -0,0 +1,268 @@
+                                            (embeddings2 - mean) / stddev,
+                                            distance_metric)
+        best_threshold = _calculate_best_threshold(thresholds,
+                                                   dist[train_set],
+                                                   labels[train_set])
+        predictions = np.less(dist[test_set], best_threshold)
+        accuracy[fold_idx] = accuracy_score(labels[test_set], predictions)
+        recall[fold_idx] = recall_score(labels[test_set], predictions)
+        precision[fold_idx] = precision_score(labels[test_set], predictions)
+    return accuracy, recall, precision
+
+
+def _parse_arguments():
+    parser = ArgumentParser()
+    parser.add_argument('--image_dir',
+                        type=str,
+                        required=True,
+                        help='Path to the image directory.')
+    parser.add_argument('--pairs_file_name',
+                        type=str,
+                        required=True,
+                        help='Filename of pairs.txt')
+    parser.add_argument('--face_verification_container',
+                        type=str,
+                        required=True,
+                        help='Name of docker container for face verification')
+    parser.add_argument('--num_folds',
+                        type=int,
+                        required=True,
+                        help='Number of cross validation folds')
+    distance_metrics = [f'{attr}'
+                        for attr in dir(DistanceMetric)
+                        if not callable(getattr(DistanceMetric, attr))
+                        and not attr.startswith("__")]
+    parser.add_argument(
+        '--distance_metric',
+        type=str,
+        required=True,
+        help=f"Distance metric for face verification: {distance_metrics}.")
+    parser.add_argument('--threshold_start',
+                        type=float,
+                        required=True,
+                        help='Start value for distance threshold.')
+    parser.add_argument('--threshold_end',
+                        type=float,
+                        required=True,
+                        help='End value for distance threshold')
+    parser.add_argument(
+        '--embedding_size',
+        type=int,
+        required=True,
+        help='Size of face vectors from face_verification_container.')
+    parser.add_argument(
+        '--threshold_step',
+        type=float,
+        required=True,
+        help='Step size for iterating in cross validation search.')
+    parser.add_argument(
+        '--subtract_mean',
+        action='store_true',
+        help=f"Subtract mean of embeddings before distance calculation.")
+    parser.add_argument(
+        '--divide_stddev',
+        action='store_true',
+        help=f"Divide embeddings by stddev before distance calculation.")
+    return parser.parse_args()
+
+
+def _main(args: Namespace) -> None:
```

Comment:
This script does a lot. I wonder if it would be simpler to divide it into two scripts, one that does the evaluation of an algorithm in terms of precision/recall/accuracy for a given threshold and a separate script that optimizes the threshold (which tbh could also be done manually plotting the metrics for a couple of values).